src/enclave/ServiceProvider/ServiceProvider.cpp

```
@@ -147,284 +151,197 @@ void ServiceProvider::export_public_key_code(const std::string &filename) {
-  // The quotes below are from this guide.
-
-  // "Generate a random EC key using the P-256 curve. This key will become Gb."
-  lc_ec256_private_t priv_key;
-  lc_ec256_public_t pub_key;
-  lc_check(lc_ecc256_create_key_pair(&priv_key, &pub_key));
-
-  // Save the group ID, the client's public ECCDH key, and the generated keypair for future steps.
-  memcpy(sp_db.gid, msg1->gid, sizeof(sgx_epid_group_id_t));
-  sp_db.g_a = msg1->g_a;
-  sp_db.g_b = pub_key;
-  sp_db.b = priv_key;
-
-  // "Derive the key derivation key (KDK) from Ga and Gb"
-  lc_ec256_dh_shared_t dh_key;
-  lc_check(lc_ecc256_compute_shared_dhkey(&priv_key, &msg1->g_a, &dh_key));
-
-  // "Derive the SMK from the KDK by performing an AES-128 CMAC on the byte sequence:
-  // 0x01 || SMK || 0x00 || 0x80 || 0x00
-  // using the KDK as the key. Note that || denotes concatenation and “SMK” is a literal string
-  // (without quotes)."
-  derive_key(&dh_key, SAMPLE_DERIVE_KEY_SMK, &sp_db.smk_key);
-  // We also precompute the same result with different strings for future messages.
-  derive_key(&dh_key, SAMPLE_DERIVE_KEY_MK, &sp_db.mk_key);
-  derive_key(&dh_key, SAMPLE_DERIVE_KEY_SK, &sp_db.sk_key);
-  derive_key(&dh_key, SAMPLE_DERIVE_KEY_VK, &sp_db.vk_key);
-
-  // "Query IAS to obtain the SigRL for the client's Intel EPID GID."
-  string sig_rl;
-  uint32_t gid;
-  memcpy(reinterpret_cast<uint8_t *>(&gid), &msg1->gid, sizeof(sgx_epid_group_id_t));
-  try {
-    if (!ias) {
-      throw std::runtime_error("process_msg1 called before ensure_ias_connection");
-    }
-    IAS_Request req(ias.get(), ias_api_version);
-    ias_check(req.sigrl(gid, sig_rl));
-  } catch (const std::runtime_error &e) {
-    if (require_attestation) {
-      throw;
+  if (memcmp(signer, signer_id_buf, signer_id_buf_size) != 0) {
+    printf("mrsigner is not equal!\n");
+    for (size_t i = 0; i < signer_id_buf_size; i++) {
+      printf("0x%x - 0x%x\n", (uint8_t)signer[i], (uint8_t)signer_id_buf[i]);
```

Instead of printf can we instead throw a runtime error? I guess if we do that here it'd require first storing everything that would be printed with `printf` and then throwing the error.